include/merlin/array_kernels.cuh

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:///www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cooperative_groups.h>
+#include "cuda_runtime.h"
+#include "thrust/device_vector.h"
+#include "thrust/execution_policy.h"
+#include "thrust/scan.h"
+#include "types.cuh"
+#include "utils.cuh"
+
+namespace nv {
+namespace merlin {
+
+template <typename T>
+__global__ void broadcast(T* data, T val, size_t n) {
```

Comment:
Why not using cudaMemset?

Reply:
I changed to cudaMemset which is faster. Broadcast if for not tileable val.